Remove Audit Rest API invokation

wrodrig opened the pull request:
As it is, when the `audit-1.0` feature is enabled, it automatically pulls rest-handler dependecies. 

When this new audit feature is enabled, the rest-handler dependecies will only be added when is enabling features that enable restHandler-1.0 functionalities. 

This PR is looking to address #27475

helyarp reviewed (commented):
Looks okay. Maybe update the last copyr year.

posmith commented:
Support message review not required.

jhanders34 commented on dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.audit2.0.feature:
```diff
@@ -0,0 +1,14 @@
+-bundles=com.ibm.ws.request.probe.audit.rest
+WLP-Activation-Type: parallel
+kind=ga
+edition=core
```
```suggestion
kind=noship
edition=full
```
This feature will not be put directly to ga.  It needs to get approvals done first.  Also the name of this auto feature should be changed to something like `io.openliberty.audit2.0.restHandler`

jhanders34 commented on dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.security.audit2.0.rest.feature:
```diff
@@ -0,0 +1,15 @@
+
+WLP-Activation-Type: parallel
+kind=ga
+edition=core
```
```suggestion
kind=noship
edition=full
```

jhanders34 commented on dev/com.ibm.websphere.appserver.features/visibility/protected/com.ibm.websphere.appserver.auditCollector-2.0.feature:
```diff
@@ -0,0 +1,15 @@
+  io.openliberty.auditCollector1.0.internal.ee-6.0; ibm.tolerates:="9.0"
+
+kind=ga
+edition=core
```
```suggestion
kind=noship
edition=full
```

jhanders34 commented on dev/com.ibm.websphere.appserver.features/visibility/public/audit-2.0/com.ibm.websphere.appserver.audit-2.0.feature:
```diff
@@ -0,0 +1,16 @@
+-bundles=com.ibm.ws.security.audit.utils, \
+  com.ibm.ws.security.audit.file, \
+  com.ibm.json4j
+kind=ga
+edition=core
```
```suggestion
kind=noship
edition=full
```

jhanders34 replied:
Not resolved

jhanders34 replied:
Also you will need to remove the json4j bundle.  It is now in the kernel after the CVE reporting function was put in.

wrodrig replied:
now it's fixed. I also removed the json4j bundle.

jhanders34 commented on dev/com.ibm.websphere.appserver.features/visibility/protected/com.ibm.websphere.appserver.auditCollector-2.0.feature:
```diff
@@ -0,0 +1,15 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+symbolicName=com.ibm.websphere.appserver.auditCollector-2.0
+singleton=true
+WLP-DisableAllFeatures-OnConflict: false
```
```suggestion
```

Since it is a new feature it should disable all features on conflict.

wrodrig replied:
changing to true

jhanders34 commented on dev/com.ibm.websphere.appserver.features/visibility/public/audit-2.0/com.ibm.websphere.appserver.audit-2.0.feature:
```diff
@@ -0,0 +1,16 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+symbolicName=com.ibm.websphere.appserver.audit-2.0
+WLP-DisableAllFeatures-OnConflict: false
```
```suggestion
```

Since it is a new feature it should disable all features on conflict.

wrodrig replied:
changing to true

jhanders34 commented on dev/com.ibm.ws.ejbcontainer.security.jacc_fat.2.1/fat/src/com/ibm/ws/ejbcontainer/security/jacc_fat/FATSuite.java:
```diff
@@ -60,6 +63,6 @@ public class FATSuite {
      */
     @ClassRule
-    public static RepeatTests repeat = RepeatTests.with(new JakartaEE10Action());
+    public static RepeatTests repeat = RepeatTests.with(new JakartaEE10Action()).andWith(new FeatureReplacementAction(AuditCommonTest.AUDIT_1_0_FEATURES, AuditCommonTest.AUDIT_2_0_FEATURES));
```
The comment doesn't match rule.  Should audit 2.0 tests be in full mode?

jhanders34 commented on dev/com.ibm.ws.ejbcontainer.security.jacc_fat.2/fat/src/com/ibm/ws/ejbcontainer/security/jacc_fat/FATSuite.java:
```diff
@@ -74,5 +76,5 @@ public class FATSuite {
      */
     @ClassRule
-    public static RepeatTests repeat = RepeatTests.with(new EmptyAction().fullFATOnly()).andWith(new JakartaEE9Action().removeFeatures(EE78_FEATURES).addFeatures(EE9_FEATURES).conditionalFullFATOnly(FeatureReplacementAction.GREATER_THAN_OR_EQUAL_JAVA_11)).andWith(new JakartaEE10Action().removeFeatures(EE78_FEATURES).removeFeatures(EE9_FEATURES).addFeatures(EE10_FEATURES));
+    public static RepeatTests repeat = RepeatTests.with(new EmptyAction().fullFATOnly()).andWith(new FeatureReplacementAction(AuditCommonTest.AUDIT_1_0_FEATURES, AuditCommonTest.AUDIT_2_0_FEATURES)).andWith(new JakartaEE9Action().removeFeatures(EE78_FEATURES).addFeatures(EE9_FEATURES).conditionalFullFATOnly(FeatureReplacementAction.GREATER_THAN_OR_EQUAL_JAVA_11)).andWith(new JakartaEE10Action().r ...
```
Should the audit 2.0 tests be run in full mode instead of light mode?  Right now the EE 9 and EE 10 tests will run with audit 2.0 as well and as such won't test audit 1.0 any longer.  I do wonder if not all of the tests use the audit feature it might makes sense to add a ClassRule to just the tests that use audit a and then you would get the coverage for just those tests.  Otherwise it is waste to rerun the same time without any changes if audit isn't used by the test.

wrodrig replied:
I reviewed the test bucket and seems like audit is not being tested here. I will remove the repeat.

jhanders34 commented on dev/com.ibm.ws.ejbcontainer.security.jacc_fat/bnd.bnd:
```diff
@@ -19,8 +19,8 @@ src: \
-	jacc-1.5, expressionLanguage-4.0, appsecurity-4.0, servlet-5.0, cdi-3.0, enterpriseBeansLite-4.0, appAuthorization-2.0, \
-	expressionLanguage-5.0, appsecurity-5.0, servlet-6.0, cdi-4.0, appAuthorization-2.1, jsonp-2.1
+	restconnector-2.0, audit-2.0, jacc-1.5, expressionLanguage-4.0, appsecurity-4.0, servlet-5.0, cdi-3.0, enterpriseBeansLite-4.0, appAuthorization-2.0, \
+	expressionLanguage-5.0, appsecurity-5.0, servlet-6.0, cdi-4.0, appAuthorization-2.1, jsonp-2.1, audit-2.0, restConnector-2.0
```
```suggestion
	restconnector-2.0, audit-2.0, jacc-1.5, expressionLanguage-4.0, appsecurity-4.0, servlet-5.0, cdi-3.0, enterpriseBeansLite-4.0, appAuthorization-2.0, \
	expressionLanguage-5.0, appsecurity-5.0, servlet-6.0, cdi-4.0, appAuthorization-2.1, jsonp-2.1
```
Don't need to include it twice.

jhanders34 commented on dev/com.ibm.ws.ejbcontainer.security.jacc_fat/publish/servers/com.ibm.ws.ejbcontainer.security.jacc_fat.audit/server.xml:
```diff
@@ -14,9 +14,9 @@
 
 	<featureManager>
 		<feature>servlet-3.1</feature>
```
I would undo this change.  No need to leave white space only changes in.

jhanders34 commented on dev/com.ibm.ws.ejbcontainer.security.jacc_fat/publish/servers/com.ibm.ws.ejbcontainer.security.jacc_fat.ejbjar.audit/server.xml:
```diff
@@ -17,7 +17,7 @@
 		<feature>appSecurity-2.0</feature>
 		<feature>usr:jaccTestProvider-1.0</feature>
-		<feature>audit-1.0</feature>		
+		<feature>audit-1.0</feature>
```
I would undo this change.  No need to leave white space only changes in.

jhanders34 commented on dev/com.ibm.ws.request.probe.audit.rest/.gitignore:
```diff
@@ -0,0 +1,27 @@
+#*
```
No need for this file.  Remove it.

jhanders34 commented on dev/com.ibm.ws.request.probe.audit.rest/.project:
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>com.ibm.ws.request.probe.audit.rest</name>
```
Since it is a new bundle / project, you should consider changing from the com.ibm.ws prefix to the io.openlibery prefix.  This request relates to [the Open Liberty naming guidelines](https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-naming-guidelines)

jhanders34 commented on dev/com.ibm.ws.request.probe.audit.rest/bnd.bnd:
```diff
@@ -0,0 +1,42 @@
+#*******************************************************************************
+-include= ~../cnf/resources/bnd/bundle.props
+
+bVersion= 1.0.0
```
```suggestion
bVersion= 1.0
```

jhanders34 commented on dev/com.ibm.ws.request.probe.audit.rest/bnd.bnd:
```diff
@@ -0,0 +1,42 @@
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
```
```suggestion
```

New files do not need this part of the header.

jhanders34 commented on dev/com.ibm.ws.request.probe.audit.rest/src/com/ibm/ws/request/probe/audit/rest/service/AuditPERestServiceImpl.java:
```diff
@@ -0,0 +1,101 @@
+package com.ibm.ws.request.probe.audit.rest.service;
```
In keeping with the naming guidelines the package should start with io.openliberty instead of com.ibm.ws.  And as I said in the bnd.bnd file, if the package isn't exported, it should be updated to have `.internal` as part of the package name.

Also copyright header missing.

jhanders34 replied:
Copyright header still missing

wrodrig replied:
ohh missed it

jhanders34 commented on dev/com.ibm.ws.request.probe.audit.rest/src/com/ibm/ws/request/probe/audit/rest/service/AuditPERestServiceImpl.java:
```diff
@@ -0,0 +1,101 @@
+
+	private static final String VARIABLE_REGISTRY_SERVICE = "variableRegistryService";
+	    private static final AtomicServiceReference<VariableRegistry> variableRegistryServiceRef = new AtomicServiceReference<VariableRegistry>(VARIABLE_REGISTRY_SERVICE);
+		private final ConcurrentHashMap<String, List<Map<String, Object>>> handlerEventsMap = new ConcurrentHashMap<String, List<Map<String, Object>>>();
```
```suggestion
```

Not needed.  Isn't used.

jhanders34 commented on dev/com.ibm.ws.request.probe.audit.rest/src/com/ibm/ws/request/probe/audit/rest/service/AuditPERestServiceImpl.java:
```diff
@@ -0,0 +1,101 @@
+        Tr.info(tc, "AUDIT_REST_SERVICE_STARTING");
+        variableRegistryServiceRef.activate(cc);
+
+    }
```
```suggestion
    protected void activate(ComponentContext cc) {
    }
```
Maybe just add a debug statement in the activate.  I removed the Tr.info since i didn't see a message file update for a new message.

jhanders34 commented on dev/com.ibm.ws.request.probe.audit.rest/src/com/ibm/ws/request/probe/audit/rest/service/AuditPERestServiceImpl.java:
```diff
@@ -0,0 +1,101 @@
+
+    @Modified
+    protected void modified(Map<String, Object> configuration) {}
+
```
```suggestion
```

No value in having these methods.  If you keep the deactivate method, adding a trace statement there would be ok.  Since you don't use the reason you can remove it and just have it match the parameters from the activate method.

jhanders34 commented on dev/com.ibm.ws.request.probe.audit.servlet/src/com/ibm/ws/request/probe/audit/servlet/AuditPE.java:
```diff
@@ -226,7 +244,13 @@ public void processCounter(Event event) {
+						auditPERestServiceRef.getService().auditEventMemberMgmt01(auditServiceRef, methodParams);
+					}
+					else{
+						Tr.error(tc, "Unable to invoke SECURITY_MEMBER_MGMT_01. Please confirm that Rest Handler is enabled in the server configuration.");
```
This needs to be a translated message key, not a plain English string.

wrodrig replied:
Yes I need to work to get this updated before we merge.

jhanders34 commented on dev/com.ibm.ws.request.probe.audit.servlet/src/com/ibm/ws/request/probe/audit/servlet/AuditPE.java:
```diff
@@ -247,7 +271,12 @@ public void processCounter(Event event) {
+						auditPERestServiceRef.getService().auditEventRESTAuthz(auditServiceRef, methodParams);
+					}
+					else {
+						Tr.error(tc, "Unable to invoke SECURITY_REST_HANDLER_AUTHZ. Please confirm that Rest Handler is enabled in the server configuration.");
```
This needs to be a translated message key, not a plain English string.

wrodrig replied:
Yes I need to work to get this updated before we merge.

jhanders34 commented on dev/com.ibm.ws.request.probe.audit.rest/.classpath:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="src" path="src"/>
+	<classpathentry kind="src" path="resources"/>
```
```suggestion
```

There is no resources source path.

jhanders34 commented on dev/com.ibm.ws.request.probe.audit.rest/bnd.bnd:
```diff
@@ -0,0 +1,42 @@
+Bundle-Description: RequestProbeRest: Security Audit Request Probe Rest; version=${bVersion}
+
+Export-Package: \
+com.ibm.ws.request.probe.audit.rest.service
```
I don't think that the package needs to be exported.

jhanders34 commented on dev/com.ibm.ws.request.probe.audit.rest/src/com/ibm/ws/request/probe/audit/rest/service/package-info.java:
```diff
@@ -0,0 +1,21 @@
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
```
```suggestion
```

Not needed in new files.

jhanders34 commented on dev/com.ibm.ws.request.probe.audit.rest/src/com/ibm/ws/request/probe/audit/rest/service/package-info.java:
```diff
@@ -0,0 +1,21 @@
+ * @version 1.0.0
+ */
+@org.osgi.annotation.versioning.Version("1.0.0")
+@TraceOptions(traceGroup = "requestProbe", messageBundle = "com.ibm.ws.request.probe.internal.resources.LoggingMessages")
```
```suggestion
@TraceOptions(traceGroup = "requestProbe")
```
No messages file for this bundle.

jhanders34 commented on dev/com.ibm.ws.request.probe.audit.servlet/src/com/ibm/ws/request/probe/audit/servlet/AuditPE.java:
```diff
@@ -5,7 +5,7 @@
  * http://www.eclipse.org/legal/epl-2.0/
  * 
- * SPDX-License-Identifier: EPL-2.0
+ * SPDX-License-Identifier: EPL-2.0x
```
```suggestion
 * SPDX-License-Identifier: EPL-2.0
```

jhanders34 commented on dev/com.ibm.ws.request.probe.audit.servlet/src/com/ibm/ws/request/probe/audit/servlet/AuditPE.java:
```diff
@@ -93,18 +95,25 @@ public class AuditPE implements ProbeExtension {
 
 	@Reference(service = AuditService.class, 
 			   name = KEY_AUDIT_SERVICE)
+
```
```suggestion
```

jhanders34 commented on dev/com.ibm.ws.request.probe.audit.servlet/src/com/ibm/ws/request/probe/audit/servlet/AuditPERestService.java:
```diff
@@ -0,0 +1,14 @@
+package com.ibm.ws.request.probe.audit.servlet;
```
```suggestion
/*******************************************************************************
 * Copyright (c) 2024 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *******************************************************************************/
package com.ibm.ws.request.probe.audit.servlet;
```

jhanders34 commented on dev/com.ibm.ws.request.probe.audit.servlet/src/com/ibm/ws/request/probe/audit/servlet/AuditPERestService.java:
```diff
@@ -0,0 +1,14 @@
+
+    String AUDIT_PE_REST_SOURCE_NAME = "audit_pe_rest";
+    String AUDIT_PE_REST_FILE_HANDLER_NAME = "AuditPERESTFileHandler";
+
```
```suggestion
```

I believe these are unused.

jhanders34 commented on dev/com.ibm.ws.request.probe.audit.servlet/src/com/ibm/ws/request/probe/audit/servlet/AuditPERestService.java:
```diff
@@ -0,0 +1,14 @@
+    String AUDIT_PE_REST_SOURCE_NAME = "audit_pe_rest";
+    String AUDIT_PE_REST_FILE_HANDLER_NAME = "AuditPERESTFileHandler";
+
+    void auditEventRESTAuthz(AtomicServiceReference<AuditService> auditServiceRef, Object[] methodParams);
+    void auditEventMemberMgmt01(AtomicServiceReference<AuditService> auditServiceRef, Object[] methodParams);
```
```suggestion
    void auditEventRESTAuthz(AuditService auditService, Object[] methodParams);
    void auditEventMemberMgmt01(AuditService auditService, Object[] methodParams);
```

I suggest getting the AuditService from the AtomicServiceReference and if it is not null then to call the method, but otherwise don't call the method.  Obviously this will change the logic some, but it is much cleaner.

jhanders34 commented on dev/com.ibm.ws.rest.handler.config_fat/bnd.bnd:
```diff
@@ -57,4 +61,6 @@ tested.features: componenttest-2.0,\
                  jaxrs-2.1,\
-                 restfulws-3.1
+                 restfulws-3.1,\
+                 restconnector-2.0,\
```
```suggestion
```

All tests already use restconnector-2.0 so you don't need to add it explicity to the tested features list.

jhanders34 commented on dev/com.ibm.ws.rest.handler.config_fat/fat/src/com/ibm/ws/rest/handler/config/fat/audit/ConfigRESTHandlerAuditTest.java:
```diff
@@ -0,0 +1,136 @@
+    public static RepeatTests r = EERepeatActions.repeat("com.ibm.ws.rest.handler.config.audit.fat",
+    EERepeatActions.EE10, // EE10
+    EERepeatActions.EE9, // EE9
+    EERepeatActions.EE8).with(new EmptyAction().fullFATOnly()).andWith(new FeatureReplacementAction(AuditCommonTest.AUDIT_1_0_FEATURES, AuditCommonTest.AUDIT_2_0_FEATURES));
```
I am guessing this doesn't give you what you want.  EE 10 will run with audit-1.0 and 2.0 in lite mode and EE 7 will run with audit 1.0 and 2.0 in full mode I believe.  Otherwise everything else will run just with audit 1.0.

wrodrig replied:
Thanks for catching that, I was debugging that issue.

jhanders34 commented on dev/com.ibm.ws.security.audit.source.rest/.gitignore:
```diff
@@ -0,0 +1 @@
+/dist
```
Not needed.  Remove it.

jhanders34 commented on dev/com.ibm.ws.security.audit.source.rest/src/com/ibm/ws/security/audit/source/rest/MemberManagementEvent.java:
```diff
@@ -10,7 +10,7 @@
  *     IBM Corporation - initial API and implementation
  *******************************************************************************/
-package com.ibm.ws.security.audit.event;
+package com.ibm.ws.security.audit.source.rest;
```
Looks like these two classes can just move over to the com.ibm.ws.request.probe.audit.rest project.  Looks like it is the only user of these classes.

wrodrig replied:
Should we keep them here to make them consistent the other events?

jhanders34 reviewed (changes requested):
I didn't update every file that had the same issues, so if you see a comment, you may want to also look at the other files to see if the problem is there as well.  Additional things to look at / fix:

1. Update copyright date in files
2. Update your IDE to use 4 spaces for indentation instead of tabs.  It really makes it hard to read the code when you have 8 character tabs.
3. Instead of repeating the FATSuite with audit-2.0, it will be better to do it for the individual test especially when only 1 or 2 tests in a FATSuite actually test with audit.  Doing the repeat has no value on those tests that aren't using the audit-1.0 feature.
4. I don't think it is right to add the restConnector feature to the list of features when switching from audit 1.0 to 2.0.  I think you should just add audit-2.0 in the FeatureReplacementAction.
5. I think it is overkill to add in the audit common.tooling for doing the FeatureReplacementAction.  In light of `#4` it is just audit-1.0 to audit-2.0.  No reason to bring in the common.tooling for constants to be used for the FeatureReplacementAction.

jhanders34 commented on dev/com.ibm.ws.rest.handler.config_fat/fat/src/com/ibm/ws/rest/handler/config/fat/audit/ConfigRESTHandlerAuditTest.java:
```diff
@@ -0,0 +1,142 @@
+    /**
+    * Need the first repeat to make sure that audit-2.0 from a previous repeat gets put back to audit-1.0
+    */
+    public static RepeatTests auditRepeat = RepeatTests.with(new FeatureReplacementAction("audit-2.0", "audit-1.0"))
```
The RuleChain isn't here like it should be.

jhanders34 commented on dev/com.ibm.ws.rest.handler.config_fat/fat/src/com/ibm/ws/rest/handler/config/fat/FATSuite.java:
```diff
@@ -23,11 +23,13 @@
-                ConfigRESTHandlerJMSTest.class,
-                ConfigRESTHandlerTest.class,
-                ConfigOpenApiSchemaTest.class
+                // ConfigRESTHandlerAppDefinedResourcesTest.class,
+                // ConfigRESTHandlerJCATest.class,
+                // ConfigRESTHandlerJMSTest.class,
+                ConfigRESTHandlerAuditTest.class,
+                ConfigRestHandlerAuditFeatureTest.class,
+                // ConfigRESTHandlerTest.class,
+                // ConfigOpenApiSchemaTest.class
```
Instead of commenting out tests, if you only want to run a specific test, you may want to switch to using `-Dfat.test.class.name=<class name>` as discussed in [the Open Liberty wiki](https://github.com/OpenLiberty/open-liberty/wiki/FAT-Tests#to-run-only-a-single-test-class-in-a-fat-use-the-command)

helyarp reviewed (commented):
Looks good!

wrodrig commented:
#build

jhanders34 commented on dev/com.ibm.ws.request.probe.audit.rest/src/io/openliberty/request/probe/audit/rest/service/internal/AuditPERestServiceImpl.java:
```diff
@@ -0,0 +1,84 @@
+public class AuditPERestServiceImpl implements AuditPERestService {
+
+
+		private static final TraceComponent tc = Tr.register(AuditPERestServiceImpl.class);
```
This file is still using tabs everywhere.  It really messes up the format and makes things wrap when viewing in github when it doesn't need to.

jhanders34 replied:
Also you are registering a trace component, but never using it and you have instrumentation disabled so nothing will be injected either.  I suggested adding something to the activate if you want something to show things being activated for debug purposes.

wrodrig replied:
Removed the TraceComponent. Not needed.

jhanders34 replied:
Didn't address the tabs at all.

jhanders34 commented on dev/com.ibm.ws.request.timing.monitor_fat/fat/src/com/ibm/ws/request/timing/monitor/fat/FATSuite.java:
```diff
@@ -24,5 +24,5 @@
 
 public class FATSuite {
-
+    
```
```suggestion

```
No need for white space only changes.

jhanders34 commented on dev/com.ibm.ws.ejbcontainer.security.jacc_fat/fat/src/com/ibm/ws/ejbcontainer/security/jacc_fat/audit/EJBJarX03JACCAuditTest.java:
```diff
@@ -48,6 +55,13 @@
 @RunWith(FATRunner.class)
 public class EJBJarX03JACCAuditTest extends EJBAnnTestBase {
 
+    @ClassRule
+    public static RepeatTests r = EERepeatActions.repeat(Constants.SERVER_EJBJAR_AUDIT,
+    EERepeatActions.EE10, // EE10
+    EERepeatActions.EE9, // EE9
+    EERepeatActions.EE8).with(new EmptyAction().fullFATOnly())
+    .andWith(new FeatureReplacementAction(AuditCommonTest.AUDIT_1_0_FEATURES, AuditCommonTest.AUDIT_2_0_FEATURES));
```
```suggestion
    /**
     * Need the first repeat to make sure that audit-2.0 from a previous repeat gets put back to audit-1.0
     */
    @ClassRule
    public static RepeatTests auditRepeat = RepeatTests.with(new FeatureReplacementAction("audit-2.0", "audit-1.0").fullFATOnly())
                    .andWith(new FeatureReplacementAction("audit-1.0", "audit-2.0"));
```
Only need the audit repeat because the Jakarta repeat is already happening in FATSuite.

jhanders34 commented on dev/com.ibm.ws.ejbcontainer.security.jacc_fat/fat/src/com/ibm/ws/ejbcontainer/security/jacc_fat/audit/EJBJarX03JACCAuditTest.java:
```diff
@@ -32,6 +38,7 @@
 import componenttest.custom.junit.runner.FATRunner;
 import componenttest.custom.junit.runner.Mode;
 import componenttest.custom.junit.runner.Mode.TestMode;
+import com.ibm.ws.security.audit.fat.common.tooling.AuditCommonTest;
```
```suggestion
```

jhanders34 commented on dev/com.ibm.ws.ejbcontainer.security.jacc_fat/fat/src/com/ibm/ws/ejbcontainer/security/jacc_fat/audit/PureAnnA08JAASLoginFromEJBJACCAuditTest.java:
```diff
@@ -49,6 +56,12 @@
 public class PureAnnA08JAASLoginFromEJBJACCAuditTest extends EJBAnnTestBase {
 
     protected static Class<?> logClass = PureAnnA08JAASLoginFromEJBJACCAuditTest.class;
+    @ClassRule
+    public static RepeatTests r = EERepeatActions.repeat(Constants.SERVER_EJB_AUDIT,
+    EERepeatActions.EE10, // EE10
+    EERepeatActions.EE9, // EE9
+    EERepeatActions.EE8).with(new EmptyAction().fullFATOnly())
+    .andWith(new FeatureReplacementAction(AuditCommonTest.AUDIT_1_0_FEATURES, AuditCommonTest.AUDIT_2_0_FEATURES));
```
```suggestion
    /**
     * Need the first repeat to make sure that audit-2.0 from a previous repeat gets put back to audit-1.0
     */
    @ClassRule
    public static RepeatTests auditRepeat = RepeatTests.with(new FeatureReplacementAction("audit-2.0", "audit-1.0").fullFATOnly())
                    .andWith(new FeatureReplacementAction("audit-1.0", "audit-2.0"));
```
Only need the audit repeat because the Jakarta repeat is already happening in FATSuite.

jhanders34 commented on dev/com.ibm.ws.ejbcontainer.security.jacc_fat/fat/src/com/ibm/ws/ejbcontainer/security/jacc_fat/audit/PureAnnA08JAASLoginFromEJBJACCAuditTest.java:
```diff
@@ -32,6 +38,7 @@
 import componenttest.custom.junit.runner.FATRunner;
 import componenttest.custom.junit.runner.Mode;
 import componenttest.custom.junit.runner.Mode.TestMode;
+import com.ibm.ws.security.audit.fat.common.tooling.AuditCommonTest;
```
```suggestion
```

jhanders34 commented on dev/com.ibm.ws.ejbcontainer.security.jacc_fat/fat/src/com/ibm/ws/ejbcontainer/security/jacc_fat/audit/PureAnnA08JAASLoginFromEJBJACCAuditTest.java:
```diff
@@ -20,6 +20,12 @@
 
+import org.junit.ClassRule;
+import componenttest.rules.repeater.RepeatTests;
+import componenttest.rules.repeater.EERepeatActions;
+import componenttest.rules.repeater.EmptyAction;
```
```suggestion
```

jhanders34 commented on dev/com.ibm.ws.ejbcontainer.security.jacc_fat/fat/src/com/ibm/ws/ejbcontainer/security/jacc_fat/audit/EJBJarX03JACCAuditTest.java:
```diff
@@ -19,6 +19,12 @@
 
+import org.junit.ClassRule;
+import componenttest.rules.repeater.RepeatTests;
+import componenttest.rules.repeater.EERepeatActions;
+import componenttest.rules.repeater.EmptyAction;
```
```suggestion
```

jhanders34 commented on dev/com.ibm.ws.logstash.collector_fat/bnd.bnd:
```diff
@@ -34,4 +35,5 @@ fat.test.container.images: elastic/logstash:7.16.3
     com.ibm.ws.logging;version=latest,\
     httpunit:httpunit;version=1.5.4,\
     org.json:json;version=20080701,\
+    com.ibm.ws.security.audit_fat.common.tooling;version=latest,\
```
```suggestion
```

jhanders34 commented on dev/com.ibm.ws.logstash.collector_fat/build.gradle:
```diff
@@ -12,7 +12,8 @@
 
 dependencies {
-  requiredLibs 'org.json:json:20080701'
+  requiredLibs 'org.json:json:20080701',
+  project(':com.ibm.ws.security.audit_fat.common.tooling')
```
```suggestion
  requiredLibs 'org.json:json:20080701
```

jhanders34 commented on dev/com.ibm.ws.logstash.collector_fat/build.gradle:
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2020 IBM Corporation and others.
+ * Copyright (c) 2024 IBM Corporation and others.
```
```suggestion
 * Copyright (c) 2020 IBM Corporation and others.
```

jhanders34 commented on dev/com.ibm.ws.logstash.collector_fat/fat/src/com/ibm/ws/logstash/collector/tests/LogstashSSLTest.java:
```diff
@@ -17,6 +17,12 @@
 import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertTrue;
 
+import org.junit.ClassRule;
```
I would undo all of the changes here and add in a audit-2.0 server xml and duplicate the testLogstashForAuditEvent since that is the only test that actually does audit-1.0.  No reason to run repeat the other tests since they don't use audit.

wrodrig replied:
Done, thanks for the suggestion.

jhanders34 commented on dev/com.ibm.ws.request.timing.monitor_fat/bnd.bnd:
```diff
@@ -1,5 +1,5 @@
 #*******************************************************************************
-# Copyright (c) 2019, 2023 IBM Corporation and others.
+# Copyright (c) 2019 IBM Corporation and others.
```
```suggestion
# Copyright (c) 2019, 2023 IBM Corporation and others.
```

jhanders34 commented on dev/com.ibm.ws.request.timing.monitor_fat/bnd.bnd:
```diff
@@ -17,6 +17,8 @@ src: \
 fat.project: true
 
 tested.features: \
+    restconnector-2.0,\
+	audit-2.0,\
```
```suggestion
```

jhanders34 commented on dev/com.ibm.ws.request.timing.monitor_fat/bnd.bnd:
```diff
@@ -42,6 +44,7 @@ tested.features: \
 	com.ibm.ws.request.timing;version=latest,\
 	com.ibm.ws.request.timing.monitor;version=latest,\
 	com.ibm.ws.monitor;version=latest,\
+	com.ibm.ws.security.audit_fat.common.tooling,\
```
```suggestion
```

jhanders34 commented on dev/com.ibm.ws.rest.handler.config_fat/fat/src/com/ibm/ws/rest/handler/config/fat/audit/ConfigRESTHandlerAuditTest.java:
```diff
@@ -0,0 +1,139 @@
+    protected String auditLogFilePath = null;
+    public static final String DEFAULT_AUDIT_LOG = "audit.log";
+
+    /**
+     * Need the first repeat to make sure that audit-2.0 from a previous repeat gets put back to audit-1.0
+     */
+    @ClassRule
+    public static RepeatTests r = EERepeatActions.repeat("com.ibm.ws.rest.handler.config.audit.fat",
+                                                         EERepeatActions.EE10, // EE10
+                                                         EERepeatActions.EE9, // EE9
+                                                         EERepeatActions.EE8)
+                    .with(new FeatureReplacementAction(AuditCommonTest.AUDIT_1_0_FEATURES, AuditCommonTest.AUDIT_2_0_REST_FEATURES))
+                    .andWith(new FeatureReplacementAction("audit-1.0", "audit-2.0"));
```
```suggestion
    public static RepeatTests r = EERepeatActions.repeat("com.ibm.ws.rest.handler.config.audit.fat",
                                                         EERepeatActions.EE10, // EE10
                                                         EERepeatActions.EE9, // EE9
                                                         EERepeatActions.EE8));

    /**
     * Need the first repeat to make sure that audit-2.0 from a previous repeat gets put back to audit-1.0
     */
    public static RepeatTests auditRepeat = RepeatTests.with(new FeatureReplacementAction("audit-2.0", "audit-1.0"))
                    .andWith(new FeatureReplacementAction("audit-1.0", "audit-2.0"));

    @ClassRule
    public static RuleChain chain = RuleChain.outerRule(r).around(auditRepeat);
```

jhanders34 commented on dev/com.ibm.ws.rest.handler.config_fat/fat/src/com/ibm/ws/rest/handler/config/fat/audit/ConfigRESTHandlerAuditTest.java:
```diff
@@ -0,0 +1,139 @@
+import com.ibm.websphere.simplicity.log.Log;
+import com.ibm.ws.rest.handler.config.fat.FATSuite;
+import com.ibm.ws.security.audit.fat.common.tooling.AuditAsserts;
+import com.ibm.ws.security.audit.fat.common.tooling.AuditCommonTest;
```
```suggestion
```

jhanders34 commented on dev/com.ibm.ws.security.jaspic_fat/fat/src/com/ibm/ws/security/jaspic11/fat/audit/JASPIBasicAuthenticationAuditTest.java:
```diff
@@ -56,6 +64,10 @@
 @Mode(TestMode.FULL)
 public class JASPIBasicAuthenticationAuditTest extends JASPITestBase {
 
+    @ClassRule
+    public static RepeatTests r = EERepeatActions.repeat("com.ibm.ws.security.jaspic11.fat.audit",
+    EERepeatActions.EE11).with(new EmptyAction().fullFATOnly()).andWith(new FeatureReplacementAction(AuditCommonTest.AUDIT_1_0_FEATURES, AuditCommonTest.AUDIT_2_0_FEATURES));
+
```
```suggestion
    /**
     * Need the first repeat to make sure that audit-2.0 from a previous repeat gets put back to audit-1.0
     */
    @ClassRule
    public static RepeatTests auditRepeat = RepeatTests.with(new FeatureReplacementAction("audit-2.0", "audit-1.0"))
                    .andWith(new FeatureReplacementAction("audit-1.0", "audit-2.0"));

```
Don't need the EE repeat since it is already done in FATSuite.

jhanders34 commented on dev/com.ibm.ws.security.jaspic_fat/fat/src/com/ibm/ws/security/jaspic11/fat/audit/JASPIBasicAuthenticationAuditTest.java:
```diff
@@ -40,6 +46,8 @@
 import componenttest.topology.impl.LibertyServer;
 import componenttest.topology.impl.LibertyServerFactory;
 
+import com.ibm.ws.security.audit.fat.common.tooling.AuditCommonTest;
```
```suggestion
```

jhanders34 commented on dev/com.ibm.ws.security.jaspic_fat/fat/src/com/ibm/ws/security/jaspic11/fat/audit/JASPIBasicAuthenticationAuditTest.java:
```diff
@@ -15,6 +15,12 @@
 
+import org.junit.ClassRule;
+import componenttest.rules.repeater.RepeatTests;
+import componenttest.rules.repeater.EERepeatActions;
+import componenttest.rules.repeater.EmptyAction;
```
```suggestion
```

jhanders34 commented on dev/com.ibm.ws.security.jaspic_fat/fat/src/com/ibm/ws/security/jaspic11/fat/audit/JASPICallbackBasicAuthAuditTest.java:
```diff
@@ -65,6 +71,12 @@
 @RunWith(FATRunner.class)
 @Mode(TestMode.FULL)
 public class JASPICallbackBasicAuthAuditTest extends JASPITestBase {
+    @ClassRule
+    public static RepeatTests r = EERepeatActions.repeat("com.ibm.ws.security.jaspic11.fat.callback.audit",
+    EERepeatActions.EE10, // EE10
+    EERepeatActions.EE9, // EE9
+    EERepeatActions.EE8).with(new EmptyAction().fullFATOnly())
+    .andWith(new FeatureReplacementAction(AuditCommonTest.AUDIT_1_0_FEATURES, AuditCommonTest.AUDIT_2_0_FEATURES));
```
```suggestion
    /**
     * Need the first repeat to make sure that audit-2.0 from a previous repeat gets put back to audit-1.0
     */
    @ClassRule
    public static RepeatTests auditRepeat = RepeatTests.with(new FeatureReplacementAction("audit-2.0", "audit-1.0"))
                    .andWith(new FeatureReplacementAction("audit-1.0", "audit-2.0"));
```
Don't need the EE repeat since it is already done in FATSuite.

jhanders34 commented on dev/com.ibm.ws.security.jaspic_fat/fat/src/com/ibm/ws/security/jaspic11/fat/audit/JASPICallbackBasicAuthAuditTest.java:
```diff
@@ -25,6 +25,12 @@
 
+import org.junit.ClassRule;
+import componenttest.rules.repeater.RepeatTests;
+import componenttest.rules.repeater.EERepeatActions;
+import componenttest.rules.repeater.EmptyAction;
```
```suggestion
```

jhanders34 commented on dev/com.ibm.ws.security.jaspic_fat/fat/src/com/ibm/ws/security/jaspic11/fat/audit/JASPIFormLoginAuditTest.java:
```diff
@@ -53,6 +62,10 @@
 @Mode(TestMode.FULL)
 public class JASPIFormLoginAuditTest extends JASPITestBase {
 
+    @ClassRule
+    public static RepeatTests r = EERepeatActions.repeat("com.ibm.ws.security.jaspic11.fat.audit",
+    EERepeatActions.EE11).with(new EmptyAction().fullFATOnly());
```
```suggestion
    /**
     * Need the first repeat to make sure that audit-2.0 from a previous repeat gets put back to audit-1.0
     */
    @ClassRule
    public static RepeatTests auditRepeat = RepeatTests.with(new FeatureReplacementAction("audit-2.0", "audit-1.0"))
                    .andWith(new FeatureReplacementAction("audit-1.0", "audit-2.0"));
```
Don't need the EE repeat since it is already done in FATSuite.

jhanders34 commented on dev/com.ibm.ws.security.jaspic_fat/fat/src/com/ibm/ws/security/jaspic11/fat/audit/JASPIFormLoginAuditTest.java:
```diff
@@ -40,6 +47,8 @@
 import componenttest.topology.impl.LibertyServer;
 import componenttest.topology.impl.LibertyServerFactory;
 
+import componenttest.rules.repeater.EERepeatActions;
+
```
```suggestion
```

jhanders34 commented on dev/com.ibm.ws.security.jaspic_fat/fat/src/com/ibm/ws/security/jaspic11/fat/audit/JASPIFormLoginAuditTest.java:
```diff
@@ -25,6 +25,13 @@
 
+import org.junit.ClassRule;
+import componenttest.rules.repeater.RepeatTests;
+import componenttest.rules.repeater.EERepeatActions;
+import componenttest.rules.repeater.EmptyAction;
```
```suggestion
```

jhanders34 commented on dev/com.ibm.ws.security.jaspic_fat/fat/src/com/ibm/ws/security/jaspic11/fat/audit/JASPIFormLoginJACCAuthorizationAuditTest.java:
```diff
@@ -74,6 +79,10 @@
 @Mode(TestMode.FULL)
 public class JASPIFormLoginJACCAuthorizationAuditTest extends JASPITestBase {
 
+    @ClassRule
+    public static RepeatTests r = EERepeatActions.repeat("com.ibm.ws.security.jaspic11.fat.audit",
+    EERepeatActions.EE11).with(new EmptyAction().fullFATOnly());
```
```suggestion
    /**
     * Need the first repeat to make sure that audit-2.0 from a previous repeat gets put back to audit-1.0
     */
    @ClassRule
    public static RepeatTests auditRepeat = RepeatTests.with(new FeatureReplacementAction("audit-2.0", "audit-1.0"))
                    .andWith(new FeatureReplacementAction("audit-1.0", "audit-2.0"));
```
Don't need the EE repeat since it is already done in FATSuite.

jhanders34 commented on dev/com.ibm.ws.security.jaspic_fat/fat/src/com/ibm/ws/security/jaspic11/fat/audit/JASPIFormLoginJACCAuthorizationAuditTest.java:
```diff
@@ -24,6 +24,11 @@
 import org.junit.runner.RunWith;
+import org.junit.ClassRule;
+import componenttest.rules.repeater.RepeatTests;
+import componenttest.rules.repeater.EERepeatActions;
+import componenttest.rules.repeater.EmptyAction;
```
```suggestion
```

jhanders34 commented on dev/com.ibm.ws.webcontainer.security.jacc.1.5_fat/fat/src/com/ibm/ws/webcontainer/security/jacc15/fat/audit/BasicAuthAuditAUTHZTest.java:
```diff
@@ -83,6 +89,13 @@ public class BasicAuthAuditAUTHZTest {
     // (hopefully) be reflected in the static references as well.
     private static TestName _name = new TestName();
 
+    @ClassRule
+    public static RepeatTests r = EERepeatActions.repeat("com.ibm.ws.webcontainer.security.fat.basicauth.audit",
+    EERepeatActions.EE10, // EE10
+    EERepeatActions.EE9, // EE9
+    EERepeatActions.EE8).with(new EmptyAction().fullFATOnly())
+    .andWith(new FeatureReplacementAction(AuditCommonTest.AUDIT_1_0_FEATURES, AuditCommonTest.AUDIT_2_0_FEATURES));
```
```suggestion
    /**
     * Need the first repeat to make sure that audit-2.0 from a previous repeat gets put back to audit-1.0
     */
    @ClassRule
    public static RepeatTests auditRepeat = RepeatTests.with(new FeatureReplacementAction("audit-2.0", "audit-1.0"))
                    .andWith(new FeatureReplacementAction("audit-1.0", "audit-2.0"));
```
Don't need the EE repeat since it is already done in FATSuite.

jhanders34 commented on dev/com.ibm.ws.webcontainer.security.jacc.1.5_fat/fat/src/com/ibm/ws/webcontainer/security/jacc15/fat/audit/BasicAuthAuditAUTHZTest.java:
```diff
@@ -16,6 +16,12 @@
 
+import org.junit.ClassRule;
+import componenttest.rules.repeater.RepeatTests;
+import componenttest.rules.repeater.EERepeatActions;
+import componenttest.rules.repeater.EmptyAction;
```
```suggestion
```

jhanders34 commented on dev/io.openliberty.checkpoint_fat/build.gradle:
```diff
@@ -10,6 +10,10 @@
  * Contributors:
  *     IBM Corporation - initial API and implementation
  *******************************************************************************/
+dependencies {
+    project(':com.ibm.ws.security.audit_fat.common.tooling')
+}
+
```
```suggestion
```

jhanders34 commented on dev/io.openliberty.checkpoint_fat/build.gradle:
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2022 IBM Corporation and others.
+ * Copyright (c) 2022, 2024 IBM Corporation and others.
```
```suggestion
 * Copyright (c) 2022 IBM Corporation and others.
```

jhanders34 commented on dev/io.openliberty.checkpoint_fat/bnd.bnd:
```diff
@@ -152,6 +153,7 @@ tested.features: \
 	io.openliberty.jakarta.restfulWS.3.0,\
 	jakarta.ws.rs-api,\
 	io.openliberty.jakarta.cdi.3.0,\
+	com.ibm.ws.security.audit_fat.common.tooling,\
```
```suggestion
```

jhanders34 commented on dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/AuditTest.java:
```diff
@@ -52,7 +54,8 @@ public class AuditTest extends FATServletClient {
     @ClassRule
```
```suggestion
```

jhanders34 commented on dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/AuditTest.java:
```diff
@@ -52,7 +54,8 @@ public class AuditTest extends FATServletClient {
     public static RepeatTests r = RepeatTests.withoutModification()
                     .andWith(new JakartaEE9Action().forServers(SERVER_NAME).fullFATOnly())
-                    .andWith(new JakartaEE10Action().forServers(SERVER_NAME).fullFATOnly());
+                    .andWith(new JakartaEE10Action().forServers(SERVER_NAME).fullFATOnly())
+                    .andWith(new FeatureReplacementAction(AuditCommonTest.AUDIT_1_0_FEATURES, AuditCommonTest.AUDIT_2_0_FEATURES));
```
```suggestion
                    .andWith(new JakartaEE10Action().forServers(SERVER_NAME).fullFATOnly()));

    /**
     * Need the first repeat to make sure that audit-2.0 from a previous repeat gets put back to audit-1.0
     */
    public static RepeatTests auditRepeat = RepeatTests.with(new FeatureReplacementAction("audit-2.0", "audit-1.0"))
                    .andWith(new FeatureReplacementAction("audit-1.0", "audit-2.0"));

    @ClassRule
    public static RuleChain chain = RuleChain.outerRule(r).around(auditRepeat);```

wrodrig commented:
#build

wrodrig commented:
#libby

wrodrig commented:
#build

wrodrig commented:
Pers build that fixes the builds of PB
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_zGXBAAi4Ee-9IeXbkm35aA

wrodrig commented:
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_tYA7sAu5Ee-d7o-0EQ2ULA

wrodrig commented:
#build

jhanders34 commented on dev/io.openliberty.jakartaee10.internal_fat/fat/src/io/openliberty/jakartaee10/internal/tests/EE10FeatureCompatibilityTest.java:
```diff
@@ -126,6 +126,7 @@ private static Set<String> getCompatibleFeatures(Set<String> allFeatures, boolea
         compatibleFeatures.remove("sipServlet-1.1"); // purposely not supporting EE 10
         compatibleFeatures.remove("springBoot-1.5"); // springBoot 3.0 only supports EE10
         compatibleFeatures.remove("springBoot-2.0");
+        compatibleFeatures.remove("audit-2.0");//depends on restHandler-1.0
```
```suggestion
```

jhanders34 commented on dev/io.openliberty.jakartaee10.internal_fat/fat/src/io/openliberty/jakartaee10/internal/tests/EE10FeatureCompatibilityTest.java:
```diff
@@ -176,6 +177,7 @@ public static Set<String> getIncompatibleFeatures(Set<String> allFeatures, Set<S
 
         incompatibleFeatures.remove("mpReactiveMessaging-3.0"); //still in development
         incompatibleFeatures.remove("mpTelemetry-2.0"); //Not yet assigned to an MPXX_FEATURES_ARRAY
+        incompatibleFeatures.remove("audit-2.0");//depends on restHandler-1.0
```
```suggestion
```

jhanders34 commented on dev/io.openliberty.jakartaee10.internal_fat/fat/src/io/openliberty/jakartaee10/internal/tests/JakartaEE10Test.java:
```diff
@@ -52,6 +52,7 @@ private static Set<String> getCompatFeatures(boolean openLibertyOnly) {
         compatFeatures.remove("jsonpContainer-2.1");
         compatFeatures.remove("passwordUtilities-1.1");
         compatFeatures.remove("persistenceContainer-3.1");
+        compatFeatures.remove("audit-2.0");//depends on restHandler-1.0
```
```suggestion
        compatFeatures.remove("audit-2.0");
```
It doesn't depend on restHandler-1.0 so that does seem weird to make that statement

jhanders34 commented on dev/io.openliberty.jakartaee11.internal_fat/fat/src/io/openliberty/jakartaee11/internal/tests/EE11FeatureCompatibilityTest.java:
```diff
@@ -136,6 +136,7 @@ private static Set<String> getCompatibleFeatures(Set<String> allFeatures, boolea
         compatibleFeatures.remove("mpMetrics");
 
         compatibleFeatures.remove("jwtSso-1.0"); // this will be removed when MP supports EE 11
+        compatibleFeatures.remove("audit-2.0");//depends on restHandler-1.0
```
```suggestion
```

jhanders34 commented on dev/io.openliberty.jakartaee11.internal_fat/fat/src/io/openliberty/jakartaee11/internal/tests/EE11FeatureCompatibilityTest.java:
```diff
@@ -180,6 +181,7 @@ public static Set<String> getIncompatibleFeatures(Set<String> allFeatures, Set<S
 
         incompatibleFeatures.remove("mpReactiveMessaging-3.0"); //still in development
         incompatibleFeatures.remove("mpTelemetry-2.0"); //Not yet assigned to an MPXX_FEATURES_ARRAY
+        incompatibleFeatures.remove("audit-2.0");//depends on restHandler-1.0
```
```suggestion
```

jhanders34 commented on dev/io.openliberty.jakartaee11.internal_fat/fat/src/io/openliberty/jakartaee11/internal/tests/JakartaEE11Test.java:
```diff
@@ -48,6 +48,7 @@ private static Set<String> getCompatFeatures(boolean openLibertyOnly) {
         compatFeatures.remove("jsonpContainer-2.1");
         compatFeatures.remove("passwordUtilities-1.1");
         compatFeatures.remove("persistenceContainer-3.2");
+        compatFeatures.remove("audit-2.0");//depends on restHandler-1.0
```
```suggestion
        compatFeatures.remove("audit-2.0");
```

jhanders34 commented on dev/io.openliberty.jakartaee9.internal_fat/fat/src/io/openliberty/jakartaee9/internal/tests/EE9FeatureCompatibilityTest.java:
```diff
@@ -136,7 +136,9 @@ private static Set<String> getCompatibleFeatures(Set<String> allFeatures, boolea
         compatibleFeatures.remove("mpHealth"); //versionless features in development
         compatibleFeatures.remove("mpMetrics");
-
+        
+        compatibleFeatures.remove("audit-2.0");//depends on restHandler-1.0
+        
```
```suggestion
```

jhanders34 commented on dev/io.openliberty.jakartaee9.internal_fat/fat/src/io/openliberty/jakartaee9/internal/tests/EE9FeatureCompatibilityTest.java:
```diff
@@ -178,6 +180,7 @@ public static Set<String> getIncompatibleFeatures(Set<String> allFeatures, Set<S
         incompatibleFeatures.addAll(allFeatures);
         incompatibleFeatures.removeAll(compatibleFeatures);
 
+        incompatibleFeatures.remove("audit-2.0");//depends on restHandler-1.0
```
```suggestion
```

jhanders34 commented on dev/io.openliberty.jakartaee9.internal_fat/fat/src/io/openliberty/jakartaee9/internal/tests/JakartaEE9Test.java:
```diff
@@ -53,6 +53,7 @@ private static Set<String> getCompatFeatures(boolean openLibertyOnly) {
         compatFeatures.remove("passwordUtilities-1.1");
         compatFeatures.remove("persistenceContainer-3.0");
         compatFeatures.remove("mpOpenAPI-3.1");
+        compatFeatures.remove("audit-2.0");//depends on restHandler-1.0
```
```suggestion
        compatFeatures.remove("audit-2.0");
```

jhanders34 commented on dev/io.openliberty.jakartaee10.internal_fat/fat/src/io/openliberty/jakartaee10/internal/tests/EE10FeatureCompatibilityTest.java:
```diff
@@ -126,6 +126,7 @@ private static Set<String> getCompatibleFeatures(Set<String> allFeatures, boolea
         compatibleFeatures.remove("sipServlet-1.1"); // purposely not supporting EE 10
         compatibleFeatures.remove("springBoot-1.5"); // springBoot 3.0 only supports EE10
         compatibleFeatures.remove("springBoot-2.0");
+        compatibleFeatures.remove("audit-2.0");
```
```suggestion
```

jhanders34 commented on dev/io.openliberty.jakartaee10.internal_fat/fat/src/io/openliberty/jakartaee10/internal/tests/EE10FeatureCompatibilityTest.java:
```diff
@@ -176,7 +177,7 @@ public static Set<String> getIncompatibleFeatures(Set<String> allFeatures, Set<S
         incompatibleFeatures.remove("mpReactiveMessaging-3.0"); //still in development
         incompatibleFeatures.remove("mpTelemetry-2.0"); //Not yet assigned to an MPXX_FEATURES_ARRAY
-
+        
```
```suggestion

```

jhanders34 commented on dev/io.openliberty.jakartaee11.internal_fat/fat/src/io/openliberty/jakartaee11/internal/tests/EE11FeatureCompatibilityTest.java:
```diff
@@ -136,6 +136,7 @@ private static Set<String> getCompatibleFeatures(Set<String> allFeatures, boolea
         compatibleFeatures.remove("mpMetrics");
 
         compatibleFeatures.remove("jwtSso-1.0"); // this will be removed when MP supports EE 11
+        compatibleFeatures.remove("audit-2.0");
```
```suggestion
```

jhanders34 commented on dev/io.openliberty.jakartaee9.internal_fat/fat/src/io/openliberty/jakartaee9/internal/tests/EE9FeatureCompatibilityTest.java:
```diff
@@ -136,7 +136,7 @@ private static Set<String> getCompatibleFeatures(Set<String> allFeatures, boolea
         compatibleFeatures.remove("mpHealth"); //versionless features in development
         compatibleFeatures.remove("mpMetrics");
-
+                
```
```suggestion

```

jhanders34 commented on dev/io.openliberty.jakartaee9.internal_fat/fat/src/io/openliberty/jakartaee9/internal/tests/EE9FeatureCompatibilityTest.java:
```diff
@@ -178,6 +178,7 @@ public static Set<String> getIncompatibleFeatures(Set<String> allFeatures, Set<S
         incompatibleFeatures.addAll(allFeatures);
         incompatibleFeatures.removeAll(compatibleFeatures);
 
+        incompatibleFeatures.remove("audit-2.0");
```
```suggestion
```

jhanders34 reviewed (dismissed):
I am approving the PR, but I think you need another PB with the latest changes to make sure no tests are regressed with those changes.

wrodrig commented:
#build

wrodrig commented:
Submitted new PB with release: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_FOzaUBFPEe-GLt3IcvbSMw&tab=com.ibm.team.build.web.ui.internal.editors.result.properties.PropertiesProvider

wrodrig commented:
#build

#libby

wrodrig committed:
Audit 2.0 Feature

    In this pull request we are working to add the `audit-2.0` feature. This feature is very similar to audit-1.0 feature the main difference is that it doesn't involve rest handler features. More information about this feature can be found here:

    https://github.com/OpenLiberty/open-liberty/issues/27475

wrodrig commented:
#build

#libby

wrodrig commented:
#build

#libby

wrodrig commented:
#build
#libby

wrodrig commented:
#build

jhanders34 commented on dev/io.openliberty.jakartaee10.internal_fat/fat/src/io/openliberty/jakartaee10/internal/tests/EE10Features.java:
```diff
@@ -282,6 +282,8 @@ private static Set<String> getExtendedCompatibleFeatures(Set<String> compatibleF
         // remove logAnalysis-1.0.  It depends on hpel being configured
         features.remove("logAnalysis-1.0");
 
+        feature.remove("audit-2.0");
```
```suggestion
        features.remove("audit-2.0");
```
This is what caused your build break.

jhanders34 commented:
#libby

jhanders34 commented:
#libby
